src/tools/pkcs15-init.c

```diff
@@ -1892,7 +1892,6 @@ init_gost_params(struct sc_pkcs15init_keyarg_gost_params *params, EVP_PKEY *pkey
 #else
```

Comment:
I think that `nid` should be set above as 
```
nid = EC_GROUP_get_curve_name(EC_KEY_get0_group(key));
```
so both `#endif` can remain as they were and the `switch` statement will work appropriately.